### PR TITLE
treewide: use SPDX identifier

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Switch to SPDX license identifier to allow machines to check license.

In the review process, especially openwrt/packages, we often have to comment that people should use SPDX and not the legacy "This is free software, licensed under the ...". I guess people use already existing Makefiles as reference for adding new packages. We should update all Makefiles so people will not use not up-to-date Makefiles.

### CODE REVIEW

File: package/boot/arm-trusted-firmware-mvebu/Makefile [hunk 1]
```diff
@@ -1,9 +1,8 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (aparcar) [inline package/boot/arm-trusted-firmware-mvebu/Makefile:1]: Twice?

**Review by aparcar — CHANGES_REQUESTED** [review]

**Comment by mans0n** [comment]
Right...we don't even have `/LICENSE` file anymore.

File: package/boot/arm-trusted-firmware-mvebu/Makefile (hunk 1 above)
Comment (PolynomialDivision) [reply package/boot/arm-trusted-firmware-mvebu/Makefile:1]: That was a mistake. Removed it. :)

**Comment by aparcar** [comment]
What tool did you use for this? `reuse`?

**Comment by PolynomialDivision** [comment]
> What tool did you use for this? `reuse`?

I used no tool... xD Just basic copy + paste.

**Comment by aparcar** [comment]
@PolynomialDivision this might improve your time https://github.com/fsfe/reuse-tool

**Comment by aparcar** [comment]
Actually please this tool to `lint` your changes. Also this is a bit of a massive change, I'd prefer to make this happen in smaller chunks. We could start with all the stuff in `./include`.

**Comment by neheb** [comment]
why or-later and not only?

**Comment by f00b4r0** [comment]
> > > I am unsure. We can also use `only`.
> 
> > 
> 
> > OpenWrt uses [GPL-2.0-only](https://github.com/openwrt/openwrt/blob/master/COPYING#L3) by default, we _should_ use it unless explicitly specified otherwise.
> 
> 
> 
> Thanks! :) That is a good argument.

It's not just a good argument. It's about respecting licensing and copyright. In particular your change would effectively relicense all the files that explicitly say "licensed under GPL version 2."

This is exactly why I personally believe it's a very bad idea to _replace_ license boilerplate with SPDX. Especially when the change is done by hand, which is very error-prone. The SPDX tag should only be _added_, that's far less invasive.

**Comment by PolynomialDivision** [comment]
> It's not just a good argument. It's about respecting licensing and copyright.

Yeah, sorry. I did not want to disrespect any licensing and copyright. I was just unsure if `GPL-2.0-only` or `GPL-2.0-or-later`. Obviously I chosed the wrong one. I did this also by hand since sometimes the header look different and I wanted to unify them. Also one package is licensed by `apache`. I was just bored and wanted to do something useful while watching a movie. ;)

**Comment by BKPepe** [comment]
How did the discussion end up, if I may to ask? Because we have opened pull request to the packages repository and it would be good to know the result, if we want to proceed further with that PR.

**Comment by aparcar** [comment]
The PR looks empty, que pasa?

**Comment by BKPepe** [comment]
Because it was badly rebased and no longer restored. The old commit can be found here: https://github.com/openwrt/openwrt/commit/c78a5bc49b5067471710ab972a362770bb8e9169

**Comment by aparcar** [comment]
@PolynomialDivision now that you got commit access, could you please give this another go and merge it :)?

**Comment by BKPepe** [comment]
@PolynomialDivision 👀 Kind reminder, please. This is blocking also https://github.com/openwrt/packages/pull/18875

**Pull request closed without merging**